src/analyses/dependence_graph.cpp

```diff
@@ -203,6 +203,26 @@ void dep_graph_domaint::transform(
   dependence_grapht *dep_graph=dynamic_cast<dependence_grapht*>(&ai);
   assert(dep_graph!=nullptr);
 
+  // We do not propagate control dependencies on function calls, i.e., only the
+  // entry point of a function should have a control dependency on the call
+  if(!control_deps.empty())
+  {
+    const goto_programt::const_targett &dep = *control_deps.begin();
+    if(dep->is_function_call())
+    {
+      INVARIANT(
+        std::all_of(
+          std::next(control_deps.begin()),
+          control_deps.end(),
+          [](const goto_programt::const_targett &d) {
+            return d->is_function_call();
+          }),
+        "All entries must be function calls");
+
+      control_deps.clear();
+    }
+  }
```

Comment:
I remember that I wrote this piece of code, though I'm not convinced anymore that it does what I intended. I think it was supposed to check that either all of the control deps are function calls, or none of them is. Though this only checks that if the first dep is a function call, then all of them are.

Reply:
It is not critical for me that this gets merged.  My plan is to trying to break parts of the VSD branch off and get them merged separately.  If they are no longer needed or should stay a branch that is fine.  I'm willing to follow your lead / preference on this one.

Reply:
Ah I was referring to only lines 206-223 not to the whole PR. Nevermind though, I checked again and also those changes are correct actually. So the PR can go in as is.

Reply:
Great!